coredns/tests/test_check.py

```diff
@@ -0,0 +1,170 @@
+# (C) Datadog, Inc. 2010-2017
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+import os
+import mock
+import pytest
+import requests
+import subprocess
+
+# project
+from datadog_checks.coredns import CoreDNSCheck
+from datadog_checks.utils.common import get_docker_hostname
+
+instance = {
+    'prometheus_endpoint': 'http://localhost:9153/metrics',
+}
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+COREDNS_HOST = get_docker_hostname()
+COREDNS_PORT = '9153'
+
+DOCKER_COMPOSE_ARGS = [
+    "docker-compose",
+    "-f", os.path.join(HERE, 'docker', 'docker-compose.yaml')
+]
+
+DIG_ARGS = [
+    "dig",
+    "google.com",
+    "@127.0.0.1",
+    "-p",
+    "54"
+]
+
+class MockResponse:
+    """
+    MockResponse is used to simulate the object requests.Response commonly returned by requests.get
+    """
+
+    def __init__(self, content, content_type):
+        self.content = content
+        self.headers = {'Content-Type': content_type}
+
+    def iter_lines(self, **_):
+        for elt in self.content.split("\n"):
+            yield elt
+
+    def raise_for_status(self):
+        pass
+
+    def close(self):
+        pass
+
+@pytest.fixture
+def coredns():
+    env = os.environ
+    env['COREDNS_CONFIG_FOLDER'] = os.path.join(HERE, 'docker', 'coredns')
+
+    start_coredns(env)
+    yield
+    subprocess.check_call(DOCKER_COMPOSE_ARGS + ["down"], env=env)
+
+def start_coredns(env):
+    subprocess.check_call(DOCKER_COMPOSE_ARGS + ["up", "-d"], env=env)
+
+    sys.stderr.write("Waiting for CoreDNS to boot...")
+
+    attempts = 1
+    while True:
+        if attempts >= 10:
+            subprocess.check_call(DOCKER_COMPOSE_ARGS + ["down"], env=env)
+            raise Exception("CoreDNS failed to boot...")
+
+        try:
+            res = requests.get('http://{}:{}/metrics'.format(COREDNS_HOST, COREDNS_PORT))
+            # create some metrics by using dig
+            subprocess.check_call(DIG_ARGS, env=env)
+            res.raise_for_status
+            break
+        except Exception:
+            attempts += 1
+            time.sleep(1)
+
+
+@pytest.fixture
+def dockercheck():
+    return CoreDNSCheck('coredns', {}, {}, [])
+
+@pytest.fixture
+def dockerinstance():
+    return {
+        'prometheus_endpoint': 'http://{}:{}/metrics'.format(COREDNS_HOST, COREDNS_PORT),
+    }
+
+@pytest.fixture
+def aggregator():
```

Comment:
aggregator is globally available with `datadog_checks_dev` [there](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_dev/datadog_checks/dev/plugin/plugin.py#L10) so you don't need to add it here :)